eatin-client/src/components/functions/SavedStateInRecipes.ts

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useEffect, useState } from "react";
+import { useAuth } from "../../context/auth-context";
+import { useGetUserrecipesByUserIdQuery } from "../../generated/graphql";
+import { Recipe, RecipesSection } from "../types";
+
+export function useAddIsSavedToRecipesSection(givenRecipes: RecipesSection[]) {
+    const { currentUser } = useAuth();
+    const { data: userRecipesData } = useGetUserrecipesByUserIdQuery({
+        variables: { userID: currentUser ? currentUser.uid : "" },
+    });
+    const [updatedRecipes, setNewRecipes] = useState<RecipesSection[]>([]);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const addIsSavedToRecipesSection = useCallback(() => {
+        setIsLoading(true);
```

Comment:
this takes so much time so we need `isLoading` state?
have you considered doing this logic of adding the `is_saved` property in the server?
I think it makes more sense doing the logic there.

Reply:
It doesnt really take long, you want me to remove the isloading?

And about the server thing - I did think about it but when I checked how the client gets the recipes, it uses some section query and I didnt know how it was defined and I was afriad to touch it and ruin everything DX...